File: prime-number/pascal/cmd.pas
program primeNumber;
uses SysUtils;

var
    j, i, primeNumberCount, number: longint;

begin
    if (ParamStr(1) > '') then
        primeNumberCount := StrToInt(ParamStr(1))
    else
        primeNumberCount := 100;

    number := 0;
    while primeNumberCount > 0 do
    begin
        inc(number);

        j := 0;
        for i := 1 to number do
        begin
            if (number mod i = 0) then
                inc(j);
        end;

        if (j = 2) then
            dec(primeNumberCount);
    end;

    writeln('The latest prime number: ', number);
end.

File: prime-number/r/cmd.r
args = commandArgs(trailingOnly=TRUE)

primeNumberCount = 100;
if (length(args) == 1) {
    primeNumberCount = as.numeric(args[1]);
}

number = 0;
while (primeNumberCount > 0) {
    number = number + 1;

    j = 0;
    for (i in 1:number) {
        if (number %% i == 0) {
            j = j + 1;
        }
    }

    if (j == 2) {
        primeNumberCount = primeNumberCount - 1;
    }
}

paste('The latest prime number: ', number);
cat("\n")
